docker/docker-compose.apisonator.yml

```diff
@@ -12,7 +12,7 @@ services:
     command:
       3scale_backend start -p 3000 -l "${APISONATOR_LOGFILE}"
     environment:
-      RACK_ENV: development
+      RACK_ENV: test
```

Comment:
With `RACK_ENV=development` there was an issue with loading `dotenv` (because the gem not present in the image).

With `RACK_ENV=production` there is another issue, and it's that some endpoints are meant to be used for test and development only, so they are disabled in production, which makes the tests fail, specifically [this one](https://github.com/3scale/apisonator/blob/8e7ff4f28088a0b78c173c90ad267524e532b76a/app/api/internal/events.rb#L23-L37), which only works in [test and development](https://github.com/3scale/apisonator/blob/8e7ff4f28088a0b78c173c90ad267524e532b76a/app/api/internal/internal.rb#L44-L46).

```
ThreeScale::Core::Event::.delete::with an existing id event#test_0001_returns true when deleting an existing event:
ThreeScale::Core::APIClient::APIError: ThreeScale::Core::APIClient::APIError: 404 POST /internal/events/, attributes: {:status=>"not_found", :error=>"Not found"}, response.body: {"status":"not_found","error":"Not found"}


```

Reply:
Shit, we should solve this in Apisonator, I don't think we should modify this file here. Better just set `APISONATOR_REL` to the last GA instead of `latest`.

Reply:
Another thing that makes no sense IMO is having the `.env` file tracked by git. It's OK to track a `.env.example` but not the real one.

Reply:
@jlledom IMO, changing `RACK_ENV=test` is fine, because indeed we are using apisonator here just for test, not for development, so I don't see a problem.

Also, I think using `latest` is fine, because we probably want pisoni development to be aligned with the latest upstream, because SaaS goes faster than on-premises.

Re: `.env` file, in general I don't like the current setup at all. I think it's overly complicated with a complex (at least for me) `Makefile`, several docker-compose files, several Dockerfiles with different builder images. The base images are also outdated and we need to upgrade them.
So, I would revamp the whole thing and make it simpler and more maintainable. But... that's for another PR :wink:

Reply:
I opened a JIRA: https://issues.redhat.com/browse/THREESCALE-10802
(description TBD)

Reply:
OK, fair enough